crates/amaru-ledger/src/governance.rs

```diff
@@ -13,20 +13,27 @@ impl DRepsSummary {
     pub fn new(db: &impl Snapshot) -> Result<Self, StoreError> {
         let dreps = db
             .iter_dreps()?
-            .map(|(k, _)| match k {
-                StakeCredential::AddrKeyhash(hash) => DRep::Key(hash),
-                StakeCredential::ScriptHash(hash) => DRep::Script(hash),
+            .map(|(k, Row { registered_at, .. })| {
+                let drep = match k {
+                    StakeCredential::AddrKeyhash(hash) => DRep::Key(hash),
+                    StakeCredential::ScriptHash(hash) => DRep::Script(hash),
+                };
+                (drep, registered_at)
             })
-            .collect::<BTreeSet<_>>();
+            .collect::<BTreeMap<_, _>>();
 
         let delegations = db
             .iter_accounts()?
             .filter_map(|(credential, account)| {
-                account.drep.and_then(|drep| {
-                    if !dreps.contains(&drep) {
-                        return None;
+                account.drep.and_then(|(drep, since)| {
+                    let registered_at = dreps.get(&drep);
+                    if let Some(registered_at) = registered_at {
+                        if since > *registered_at {
```

Comment:
Now is where we need to carefully check things. I would assume `>=` is okay from a slot-standpoint, because the registration can happen in the same transaction. But that means we need to include the certificate position as part of the ordering. Because certificates are ordered, and so long as the DRep is registered beforehand, then the delegation shall be authorized in the same transaction.